src/streaming/controllers/ScheduleController.js

```diff
@@ -185,22 +187,37 @@ function ScheduleController(config) {
         log('Schedule controller stopping for ' + type);
     }
 
+    function hasTopQualityChanged(type, id) {
+
+        topQualityIndex[id] = topQualityIndex[id] || {};
+        let newTopQualityIndex = abrController.getTopQualityIndexFor(type,id);
+
+        if ( topQualityIndex[id][type] != newTopQualityIndex ) {
+            log('Top quality'  + type + ' index has changed from ' + topQualityIndex[id][type] + ' to ' + newTopQualityIndex);
+            topQualityIndex[id][type] = newTopQualityIndex;
+            return true;
+        }
+        return false;
+
+    }
+
     function schedule() {
 
         if (isStopped || isFragmentProcessingInProgress || !bufferController || playbackController.isPaused() && !scheduleWhilePaused) return;
 
         validateExecutedFragmentRequest();
 
-        const isReplacement = replaceRequestArray.length > 0;
-        const readyToLoad = bufferLevelRule.execute(streamProcessor, type, streamController.isVideoTrackPresent());
+        let isReplacement, topQualityChanged, readyToLoad;
+        if ( (isReplacement = replaceRequestArray.length > 0) ||
```

Comment:
Don't do assignment inside the condition of an if statement. I'd prefer the old way with the const above, or just use the rhs of the assignment alone (topQualityChanged and readyToLoad not being used after the if)